### PR TITLE
added chat component

### PR DESCRIPTION
**Pull request by xinkin**
I created a branch from my forked repo

### CODE REVIEW

**Comment by vrajdesai78** [comment]
messages are rendering twice, can you fix it. Also, I would suggest you improve the UI like how we have on our prod app. 

Screenshot 2023-10-18 at 3 02 22 PM
Screenshot 2023-10-18 at 3 02 34 PM

**Comment by xinkin** [comment]
I have made the necessary changes: fixed username rendering twice, added SVG, shortened the width, made UI palette similar to prod, please see to it @vrajdesai78

**Comment by xinkin** [comment]
Screenshot 2023-10-21 at 4 09 24 AM
Screenshot 2023-10-21 at 4 09 10 AM
Screenshot 2023-10-21 at 4 08 54 AM
here are a few screenshots of the same.

**Comment by vrajdesai78** [comment]
> Screenshot 2023-10-21 at 4 09 24 AM Screenshot 2023-10-21 at 4 09 10 AM Screenshot 2023-10-21 at 4 08 54 AM here are a few screenshots of the same.

UI looks better now, still you need to hide it on clicking a chat button in bottom right, like how we do for peers list. Still if I send a message it is rendering twice in the first peer. 

image

**Comment by xinkin** [comment]
@vrajdesai78 I am adding the show and hide feature but regards the twice message rendering I have checked my states thoroughly and I fixed the rendering of empty strings, I am not sure why you are having the message been rendered twice, its not an issue in my machine

**Comment by vrajdesai78** [comment]
@xinkin just a minor change

Instead of reading `data:received` event in chat.tsx and adding chats, use that code inside `[roomId]/page.tsx` as shown below. Remove the event listener from `chat.tsx`.  

image

File: src/components/BottomBar/BottomBar.tsx [hunk 1]
```diff
@@ -49,36 +53,38 @@ const BottomBar: React.FC<BottomBarProps> = () => {
...
+            onClick={() => setPromptView('request-to-speak')}
+          >
+            {BasicIcons.requestToSpeak}
+            <div>Request to speak</div>
+          </OutlineButton>
+        )}
+      </div>
 
       {/* Bottom Bar Center */}
       <div className="flex items-center mr-20 gap-4">
```
Comment (vrajdesai78) [inline src/components/BottomBar/BottomBar.tsx:86]: Remove `mr-20`

**Comment by xinkin** [comment]
@vrajdesai78 made the changes please review

File: src/components/Chat/Chat.tsx [hunk 2]
```diff
@@ -46,27 +43,16 @@ const Chat = () => {
...
-      const newChatMessage = {
-        name: data.payload.name,
-        text: data.payload.message,
-        is_user: false,
-      };
-      addChatMessage(newChatMessage);
-    }
-  });
-
   console.log(chatMessages);
```
Comment (vrajdesai78) [inline src/components/Chat/Chat.tsx:46]: remove this unnecessary console logs

**Comment by xinkin** [comment]
@vrajdesai78 removed unnecessary console logs

**Comment by vrajdesai78** [comment]
Great work @xinkin, LGTM 🚀

**Pull request merged**